### PR TITLE
ClearML Integration

### LINKED ISSUE #540: Add ClearML Server Deployment Option in Qhub
ClearML is a way to track experiment results for ML Models.  It'd be nice to add to Qhub.

Looks like some instructions for deployment exist at https://allegro.ai/clearml/docs/rst/deploying_clearml/index.html.  Initially

Likely, I'll plan to use the `Helm` deployment option.

More details:
ClearML is broken into roughly 3 parts. A local python package, a ClearML server, and a ClearML Agent for ML-Ops orchestration.  The priority is the ClearML server, if the ClearML agent is not too heavy a lift that would be great too (unclear how it’s orchestration functionality integrates with QHub). 

https://github.com/allegroai/clearml/#clearml-architecture

Ideally, finish this in the next 2 weeks.

# Acceptance Criteria

 - [ ] A PR that demonstrates a successful deployment of ClearML keeping in mind this is an initial deployment.

### LINKED ISSUE #635: Upgrade Helm Terraform provider to the latest
Currently we use a pretty old version of terraform's helm provider:

https://github.com/Quansight/qhub/blob/7b36352946170d29f9d5f1a13f634ee0961a043c/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/infrastructure/versions.tf#L23

The current version is [2.1.2](https://registry.terraform.io/providers/hashicorp/helm/latest)
The current version has some issues like this one: https://github.com/hashicorp/terraform-provider-helm/issues/497
It would be great to upgrade to the latest version.

### PR DESCRIPTION
**Pull request by aktech**
Integrates ClearML Fixes #540 

- [x] Add ClearML components from [cloud-ready helm chart](https://github.com/allegroai/clearml-server-k8s/tree/master/clearml-server-cloud-ready)
- [x] Custom image for Elasticsearch (created by @Adam-D-Lewis) TODO: Move image to Quansight dockerhub
- [x] Adds docs
- [x] Adds three subdomain DNS records (TODO: Make that optional, currently adds irrespective of clearml deployed or not)
- [x] Terraform Helm Provider upgrade to `2.1.2` from `1.0` Fixes #635 
- [x] Adds a nodeSelector to select labelled nodes for ClearML (due to high memory requirement for ElasticSearch), default required label: `app: clearml`
- [x] Feature flag & ability to label nodes(so that people can label ClearML nodes): @viniciusdc 

Co-Authors:
- @Adam-D-Lewis 
- @viniciusdc

### CODE REVIEW

File: docs/source/08_integrations/2_clearml.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# ClearML
+
+ClearML Integration comes built in with QHub, here is how you would
+enable this integration.
+
```
Comment (Adam-D-Lewis) [inline docs/source/08_integrations/2_clearml.md:5]: We should mention that only GCP is supported currently.

**Comment by Adam-D-Lewis** [comment]
@viniciusdc and I pair programmed most of today to get the feature flag working. https://github.com/Quansight/qhub/pull/649

File: qhub/deploy.py [hunk 2]
```diff
@@ -111,3 +113,12 @@ def guided_install(
...
     terraform.apply(directory="infrastructure")
+
+
+def add_clearml_dns(zone_name, record_name, record_type, ip_or_hostname):
+    logger.info(f"Setting DNS record for ClearML for record: {record_name}")
+    update_record(zone_name, f"clearml.app.{record_name}", record_type, ip_or_hostname)
+    update_record(zone_name, f"clearml.api.{record_name}", record_type, ip_or_hostname)
+    update_record(
+        zone_name, f"clearml.files.{record_name}", record_type, ip_or_hostname
+    )
```
Comment (brl0) [inline qhub/deploy.py:124]: This seems like a really interesting and awesome integration with qhub, it piqued my interest enough to take a look at this PR to see how it is being integrated.

One thing I noticed, and I may be missing something here, but the docs in this PR indicate the domain as `app.clearml.your-qhub-domain.com`, which seems appropriately conventional in the pattern of `*.subdomain.domain.tld`, whereas here it appears that the parts of the domain were swapped, resulting in the seemingly unconventional pattern of `subdomain.*.domain.tld`.

My apologies for the unsolicited comment here, especially if my understanding or interpretation is incorrect, but I thought saying something might provide an opportunity to make the appropriate changes if warranted.

Anyway, keep up the great work!

Comment (aktech) [reply qhub/deploy.py:124]: Hey @brl0 that's a very valid point. I will fix that.

**Review by Adam-D-Lewis — APPROVED** [review]
LGTM

**Pull request merged**